sklearn/datasets/_openml.py

```diff
@@ -840,7 +840,7 @@ def fetch_openml(
             raise ValueError(
                 "Dataset data_id={} and version={} passed, but you can only "
                 "specify a numeric data_id or a version, not "
-                "both.".format(data_id, name))
+                "both.".format(data_id, version))
```

Comment:
May we add a new test cast  to the following function?

https://github.com/scikit-learn/scikit-learn/blob/6bfaceded8e99396ff18356c9b97a7a673b6d9e1/sklearn/datasets/tests/test_openml.py#L1218

The test would make sure that the error message has the correct data_id and version.

Reply:
Added.